src/DataWrangling/ECCO/ECCO_darwin.jl

```diff
@@ -0,0 +1,179 @@
+using MeshArrays
+using JLD2
+using Glob
+
+struct ECCO2DarwinMonthly <:SomeECCODataset end
+struct ECCO4DarwinMonthly <:SomeECCODataset end
+
+# URLs for the ECCO datasets specific to each version
+const ECCO4Darwin_url = "https://ecco.jpl.nasa.gov/drive/files/ECCO2/LLC90/ECCO-Darwin/"
+const ECCO2Darwin_url = "https://ecco.jpl.nasa.gov/drive/files/ECCO2/LLC270/ECCO-Darwin_extension/"
+
+Base.size(data::Metadata{<:ECCO4DarwinMonthly}) = (720,  360, 50, length(data.dates))
+Base.size(::Metadatum{<:ECCO4DarwinMonthly})    = (720,  360, 50, 1)
+Base.size(data::Metadata{<:ECCO2DarwinMonthly}) = (1440, 720, 50, length(data.dates))
+Base.size(::Metadatum{<:ECCO2DarwinMonthly})    = (1440, 720, 50, 1)
+
+# The whole range of dates in the different dataset datasets
+all_dates(::ECCO4DarwinMonthly, name) = DateTime(1992, 1, 1) : Month(1) : DateTime(2023, 3, 1)
+all_dates(::ECCO2DarwinMonthly, name) = DateTime(1992, 1, 1) : Month(1) : DateTime(2025, 5, 1)
+
+ECCO_Darwin_timestep(::Metadatum{<:ECCO4DarwinMonthly}) = 3600
+ECCO_Darwin_timeref(::Metadatum{<:ECCO4DarwinMonthly}) = DateTimeProlepticGregorian(1992, 1, 1, 12, 0, 0)
```

Comment:
Since we are dispatching on the dataset, its redundant to also have a unique function name. Instead this is an opportunity to create an interface for new info, eg 

```julia
function metadata_time_step end
function metadata_epoch end
```

Put the above functions at the top of the DataWrangling module. Then you can extend these inside `ECCO_darwin.jl`:

```julia
metadata_time_step(::Metadatum{<:ECCO4DarwinMonthly}) = 3600
metadata_epoch(::Metadatum{<:ECCO4DarwinMonthly}) = DateTimeProlepticGregorian(1992, 1, 1, 12, 0, 0)
```

(I'm not sure how these are used, but this is a good way to approach the software design)

Reply:
Done